flocker/node/functional/test_deploy.py

```diff
@@ -360,11 +360,13 @@ def inspect_application(_):
             deferred_list = docker_client.list()
 
             def app_memory(units):
-                unit = units.pop()
-                self.assertEqual(unit.mem_limit, EXPECTED_MEMORY_LIMIT)
-                return deferred_list
-
+                checked = [
+                    self.assertEqual(unit.mem_limit, EXPECTED_MEMORY_LIMIT)
+                    for unit in units if unit.name == application_name
+                ]
```

Comment:
To know if this has had a beneficial effect, it would be great to log something about what containers this new code is filtering out.  A log message with, say, the names of all containers listed would give us something to point at and say "Aha, there were 2 other containers and the `units.pop()` previously picked the wrong one, the assumption is validated and the bug is really fixed."